packages/actor-extract-links-extract-tree/lib/SolutionDomain.ts

```diff
@@ -0,0 +1,198 @@
+import { SolutionRange } from './SolutionRange';
+import { LogicOperator } from './solverInterfaces';
+
+const AN_EMPTY_SOLUTION_RANGE = new SolutionRange(undefined);
+/**
+ * A class representing the domain of a solution of system of boolean equation.
+ * Every operation return a new object.
+ */
+export class SolutionDomain {
+  /**
+     * The multiple segment of the domain, it is always order by the lower bound
+     * of the SolutionRange.
+     */
+  private domain: SolutionRange[] = [];
```

Comment:
There's something wrong with your use of terminology here.
Domain != range, but domain here is expressed in terms of a range.
Can you clarify? We may need to change terminology.

Reply:
I was seeing it, as a domain is a set of ranges, because it can have holes.

Reply:
Ok, then I would suggest renaming `SolutionRange` to `SolutionInterval`.
Because Domain and Range have some different connotations when used together.